public/app/features/dashboard/components/ShareModal/SharePublicDashboard.tsx

```diff
@@ -121,6 +128,7 @@ export const SharePublicDashboard = (props: Props) => {
                   label="Your entire dashboard will be public"
                   value={acknowledgements.public}
                   disabled={publicDashboardPersisted(publicDashboard)}
+                  aria-label={selectors.WillBePublicCheckbox}
```

Comment:
~~`aria-label`~~ `data-testid`

This is a good example to explain why using aria-label (as we used to) is so harmful: previously this checkbox would be labelled as "Your entire dashboard will be public". With this change it would be labelled "Public Dashboard will be public checkbox" which isn't a great label to put in the UI!